runtime.ts

```diff
@@ -1705,6 +1709,58 @@ module J2ME {
       return;
     }
   }
+
+  /**
+   * Last time we preempted a thread.
+   */
+  var lastPreemption = 0;
+
+  /**
+   * The preemption check should be quick. We don't always want to measure
+   * time so we use a quick counter and mask to determine when to do the
+   * more expensive preemption check.
+   */
+  var preemptionSamples = 0;
+  var preemptionSampleMask = 0xFF;
+
+  /**
+   * Number of ms between preemptions, chosen arbitrarily.
+   */
+  var preemptionInterval = 100;
+
+  /**
+   * Number of preemptions thus far.
+   */
+  export var preemptionCount = 0;
+
+  /**
+   * Periodically preempts the current thread. Calls to this method are inserted in
+   * methods that can yield.
+   */
+  export function checkPreemption() {
+    preemptionSamples ++;
+    if ((preemptionSamples & preemptionSampleMask) === 0) {
```

Comment:
Is there any value to combining the increment statement with the conditional check, i.e. `if ((++preemptionSamples & preemptionSampleMask) === 0)`? I know this code is performance-sensitive, but unsure how much of a difference that would make.